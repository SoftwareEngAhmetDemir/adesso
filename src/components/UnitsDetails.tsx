import { useParams } from 'react-router-dom';
import Header from './Header';
import { Fragment, useEffect } from 'react';
import { Unit } from '../entities/data-reponose';
import { action, store } from '../redux/store';
import { GET_DATA_REQUEST } from '../redux/actions';

function UnitsDetails() {
  let { id }: any = useParams();
  let state = store.getState();
  let data: Unit[] = state.data;

  useEffect(() => {
    if (data[id] === undefined) action(GET_DATA_REQUEST,null);
  }, [data, id]);

  if (data[id] === undefined) {
    return <div>loading</div>;
  } else
    return (
      <Fragment>
        <Header title="Units Details" />
        <div className="container">
          <div className="d-lg-block d-flex justify-content-center mt-3">
            <div className="row justify-content-center col-xl-4 col-lg-9 col-12">
              <div className="d-flex align-items-center col-5 border">
                <h6 className="h-50">ID</h6>
              </div>
              <div className="d-flex align-items-center col-7 border">
                {data[id].id}
              </div>
              <div className="d-flex align-items-center col-5 border">
                <h6 className="h-50">Name</h6>
              </div>
              <div className="d-flex align-items-center col-7 border">
                {data[id].name}
              </div>
              <div className="d-flex align-items-center col-5 border">
                <h6 className="h-auto">Description</h6>
              </div>
              <div className="d-flex align-items-center col-7 border">
                {data[id].description}
              </div>
              <div className="d-flex align-items-center col-5 border">
                <h6 className="h-50">Min.Required Age</h6>
              </div>
              <div className="d-flex align-items-center col-7 border">
                {data[id].age}
              </div>
              <div className="d-flex align-items-center col-5 border">
                <h6 className="h-50">Wood Cost</h6>
              </div>
              <div className="d-flex align-items-center col-7 border">
                {data[id]?.cost?.Wood}
              </div>
              <div className="d-flex align-items-center col-5 border">
                <h6 className="h-50">Food Cost</h6>
              </div>
              <div className="d-flex align-items-center col-7 border">
                {data[id]?.cost?.Food}
              </div>
              <div className="d-flex align-items-center col-5 border">
                <h6 className="h-50">Gold Cost</h6>
              </div>
              <div className="d-flex align-items-center col-7 border">
                {data[id]?.cost?.Gold}
              </div>
              <div className="d-flex align-items-center col-5 border">
                <h6 className="h-50">Build Time</h6>
              </div>
              <div className="d-flex align-items-center col-7 border">
                {data[id].build_time}
              </div>
              <div className="d-flex align-items-center col-5 border">
                <h6 className="h-50">Reload Time</h6>
              </div>
              <div className="d-flex align-items-center col-7 border">
                {data[id].reload_time}
              </div>
              <div className="d-flex align-items-center col-5 border">
                <h6 className="h-50">Hit Points</h6>
              </div>
              <div className="d-flex align-items-center col-7 border">
                {data[id].hit_points}
              </div>
              <div className="d-flex align-items-center col-5 border">
                <h6 className="h-50">Attack</h6>
              </div>
              <div className="d-flex align-items-center col-7 border">
                {data[id].attack}
              </div>
              <div className="d-flex align-items-center col-5 border">
                <h6 className="h-50">Accurancy</h6>
              </div>
              <div className="d-flex align-items-center col-7 border">
                {data[id].accuracy}
              </div>
            </div>
          </div>
        </div>
      </Fragment>
    );
}

export default UnitsDetails;
