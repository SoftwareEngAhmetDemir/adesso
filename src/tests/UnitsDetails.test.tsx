import { shallow } from 'enzyme';
import UnitsDetails from '../components/UnitsDetails';

test('should test UnitsDetails component', () => {
 const wrapper = shallow(<UnitsDetails  />);
 expect(wrapper).toMatchSnapshot();
});