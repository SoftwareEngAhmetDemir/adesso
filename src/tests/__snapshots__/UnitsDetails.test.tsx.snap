// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test UnitsDetails component 1`] = `
<div>
  loading
</div>
`;
